arangod/IResearch/IResearchDataStore.h

```diff
@@ -0,0 +1,378 @@
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
Why are these files duplicated IResearchLink?

I understand this is a renaming. There is a problem with this. 
When you merge you will not get a conflict even if IResearchLink changes

Reply:
The idea is to move all "common" part for the  inverted index and  the link here. So it will be not duplicate but move. Currently yes. Some conflicts are expected to happen,